aptos-move/framework/supra-framework/sources/poel/poel.move

```diff
@@ -0,0 +1,308 @@
+/// borrow $Supra and stake it in the Supra PoS system. The PoEL contract periodically lends $Supra tokens to users who have submitted WETH to the Intralayer 
+/// Vault and evenly distributes these tokens among various delegation pools participating in the system. 
+/// In this setup, the PoEL contract itself acts as the delegator of the $Supra tokens in the delegation pool on behalf of the users.
+/// Note:
+/// The total lended amount through the PoEL contract depends on the Oracle price submitted to the PoEL contract.
+/// If the price of the asset goes up compared with $Supra the system lends more delegating more assets 
+/// to the delegation pools and correspondingly if the price drops the system lends less, withdrawing some assets
+/// from the delegation pools. Therefore, the PoEL contract acting as the delegation agent for the users ensures the
+/// total delegated amount from the PoEL contract is equable with the total nominal value submitted to the IntraLayer vaults.
+/// When a user wants to retrieve the original ETH that was used as collateral, they can burn the iETH, which then enables
+/// them to receive the original ETH on the Ethereum blockchain.
+
+module supra_framework::poel {
+    use supra_framework::table;
+    use supra_framework::smart_table::SmartTable;
+    //Errors:
+
+    /// thrown when the allowed signer is not the caller (owner)
+    const ERR_NOT_OWNER: u64 = 0;
+
+    /// thrown when the wrong asset weight gets calculated
+    const ERR_WRONG_WEIGHT: u64 = 1;
+
+    /// thrown when the length of the supplied collaterixation weight argument vector does not match table entries
+    const ERR_WRONG_CWV_LENGTH: u64 = 2;
+
+    /// thrown when the length of the desirability score argument vector does not match table entries
+    const ERR_WRONG_DESIRABILITY_SCORE_LEN: u64 = 3;
+
+    /// thrown when the asset id supiled is not present in the TotalLiquidityTable
+    const ERR_ASSET_NOT_PRESENT: u64 = 4;
+
+    /// thrown If coefficient_X < coefficient_b
+    const ERR_COEFA_LT_COEFB: u64 = 5;
+ 
+    // Structs:
+ 
+     /// This struct contains contains the field attributes for each asset in the liquidity table
+     struct LiquidityTableItems {
+        ///Asset_name
+        asset_name: vector<u8>,
+        ///asset_supply: Total amount of the asset that has been bridged to create the iAsset.
+        asset_supply: u64,
+        ///iAsset_supply: Total supply of iAssets generated through the scheme.
+        iAsset_supply: u64,
+        ///collaterisation_rate: Specific rate indicating the collateral security of the iAsset.
+        collateralisation_rate: u64,
+        ///desired_weights: Weights reflecting the asset's strategic importance.
+        desired_weight: u64,
+        ///desirability_score: Attractiveness score of the asset.
+        desirability_score: u64,
+        ///asset_price: Current market price of the asset.
+        asset_price: u64,
```

Comment:
Current market price against what? If this information is available via oracle, then this should not be stored.

Reply:
this is against supra price

Reply:
True, we probably don't need to store the asset price in the struct.